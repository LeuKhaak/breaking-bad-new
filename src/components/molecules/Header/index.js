import React from "react";
import { Link, useLocation } from "react-router-dom";
import * as styles from "./styles.module.scss";
import logo from "../../../assets/images/logo.png";

function Header() {
  const location = useLocation().pathname;

  return (
    <header className={styles.header}>
      <div className={styles.container}>
        <Link to="/" className={styles.link}>
          <img src={logo} alt="#" />
        </Link>
        <Link to="/catalog" className={styles.link}>
          <div
            className={styles.toCatalog}
            style={location !== "/catalog" ? { background: "none" } : {}}
          >
            <span className={styles.burger}>
              <span className={styles.dash}></span>
              <span className={styles.dash}></span>
              <span className={styles.dash}></span>
            </span>
            <span className={styles.linkText}>Catalog</span>
          </div>
        </Link>
      </div>
    </header>
  );
}

export default Header;
