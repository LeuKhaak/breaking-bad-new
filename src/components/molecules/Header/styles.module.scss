@import "../../../assets/styles/styles.scss";
.header {
  .container {
    @include content;
    max-width: 74.5rem;
    display: flex;
    justify-content: space-between;
    padding: 1rem 0;
    .link {
      text-decoration: none;
      color: $mainFontColor;
      .toCatalog {
        width: 7.8rem;
        height: 2.5rem;
        padding: 0 1.25rem 0 1rem;
        background: $gold;
        border: 1.4px solid $gold;
        box-sizing: border-box;
        border-radius: 0.25rem;
        display: flex;
        align-items: center;
        .burger {
          width: 1rem;
          height: 0.625rem;
          margin-right: 0.75rem;
          display: flex;
          flex-direction: column;
          align-items: center;
          justify-content: space-between;
          .dash {
            display: block;
            width: 1rem;
            height: 0;
            border-bottom: 0.125rem solid $mainFontColor;
          }
        }
      }
    }
  }
}
