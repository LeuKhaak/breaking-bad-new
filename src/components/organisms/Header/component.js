import React from "react";
import { Link } from "react-router-dom";
import * as styles from "./styles.module.scss";
import logo from "../../../assets/images/logo.svg";
import burger from "../../../assets/icons/burger.svg";
import Btn from "../../../components/atoms/Btn";

function Header({ location }) {
  return (
    <header className={styles.header}>
      <div className={styles.container}>
        <Link to="/" className={styles.link}>
          <img src={logo} alt="#" />
        </Link>
        <Link to="/catalog" className={styles.link}>
          <Btn
            location={location}
            btnText="Catalog"
            btnStyle="toCatalog"
            icon={burger}
          />
        </Link>
      </div>
    </header>
  );
}

export default Header;
