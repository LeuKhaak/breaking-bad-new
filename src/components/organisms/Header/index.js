import React from "react";
import { Link, useLocation } from "react-router-dom";
import * as styles from "./styles.module.scss";
import { ReactComponent as Logo } from "src/assets/images/logo.svg";
import burger from "src/assets/icons/burger.svg";
import Btn from "src/components/atoms/Btn";

function Header() {
  const location = useLocation().pathname;

  return (
    <header className={styles.header}>
      <div className={styles.container}>
        <Link to="/" className={styles.link}>
          <Logo />
        </Link>
        <Link to="/catalog" className={styles.link}>
          <Btn
            location={location}
            btnText="Catalog"
            btnStyle="toCatalog"
            icon={burger}
          />
        </Link>
      </div>
    </header>
  );
}

export default Header;
