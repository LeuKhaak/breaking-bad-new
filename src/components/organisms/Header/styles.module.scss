@import "../../../assets/styles/styles.scss";
.header {
  .container {
    @include content;
    max-width: 74.5rem;
    display: flex;
    justify-content: space-between;
    align-items: center;
    padding: 1rem 0;
    .link {
      text-decoration: none;
      color: $mainFontColor;
      line-height: 0;
    }
  }
}
