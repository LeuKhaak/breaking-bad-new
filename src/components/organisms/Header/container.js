import React from "react";
import Header from "./component";
import { useLocation } from "react-router-dom";

function HeaderContainer() {
  const location = useLocation().pathname;

  return <Header location={location} />;
}

export const container = HeaderContainer;
