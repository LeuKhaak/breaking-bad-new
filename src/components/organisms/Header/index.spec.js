import React from "react";
import { shallow } from "enzyme";
import Header from "./component";

let component = null;
const setComponent = (props) => shallow(<Header {...props} />);

beforeEach(() => {
  component = setComponent();
});

describe("Header testing:", () => {
  test("is Header rendered", () => {
    component = setComponent({
      location: "/",
    });
    expect(component).toMatchSnapshot();
  });
});
