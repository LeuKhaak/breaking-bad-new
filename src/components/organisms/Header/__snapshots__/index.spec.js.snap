// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header testing: is Header rendered 1`] = `
<header>
  <div>
    <Link
      to="/"
    >
      <img
        alt="#"
        src="test-file-stub"
      />
    </Link>
    <Link
      to="/catalog"
    >
      <Btn
        btnStyle="toCatalog"
        btnText="Catalog"
        icon="test-file-stub"
        location="/"
      />
    </Link>
  </div>
</header>
`;
